contracts/SpokePool.sol

```diff
@@ -1322,8 +1370,33 @@ abstract contract SpokePool is
         emit SetHubPool(newHubPool);
     }
 
+    function _preExecuteLeafHook(address l2TokenAddress) internal virtual {
```

Comment:
I think this is a nice pattern we should already be using, where certain SpokePool children perform hooks before diving into the base executeRelayerRefund logic. By implementing an overridable hook and making sure its called before executeSlow/RelayerRefundLeaf functions, we can reduce the amount of functions that the children SpokePools have to implement

Reply:
> Isn't that what we are doing currently by making execute{Relayer,Slow}RefundRoot() virtual?

It's not virtual anymore though - it's only marked as `override`.

![virtual](https://github.com/across-protocol/contracts-v2/assets/108695806/6ee85cb0-b06b-4ffb-a482-10bd81876d7c)

> And for the use case, would this be used in the case where the merkle leaf math is somehow different per chain?

Yeah - I was wondering about something like that. I don't know how realistic it is that we'd see that.

Reply:
Its not virtual anymore because the `_preExecuteLeafHook` is now the `virtual` method. The `_execute{Relayer,Slow}RefundRoot()` are otherwise the same currently so I moved all shared logic that could be overridden in the children in to this new hook virtual method

Reply:
> Yeah - I was wondering about something like that. I don't know how realistic it is that we'd see that.

I think we can solve that if and when we ever see that but that would cause other issues if the root submitted to the HubPool results in different proofs for different chains